### DISCUSSION TITLE
Missing methods

### DISCUSSION THREAD

**Discussion by AbuAmmar072001** [pnp/pnpcore#1075] [General] [answered]
Hi, 

I am using PnP.Core version #1.8.0 in VS 2022 C#, I am trying to get user information and check if the user is active and exists in Azure AD.

The documentation mentioned this method ValidateAndEnsureUsersAsync to check users and ensures the ones that exist in Azure AD, but I cannot find the method in the pnp context.
IWeb does not contain a definition for ValidateAndEnsureUsersAsync

![image](https://user-images.githubusercontent.com/19372527/213333632-08130693-63d2-4c3b-bb38-b3a664ea67a0.png)

`using (var context = await pnpContextFactory.CreateAsync(siteUri))
                        {
                            var userList = new List<string>
                            {
                                "ValidUser1@contoso.onmicrosoft.com",
                                "ValidUser2@contoso.onmicrosoft.com",
                                "NonExistingUser@contoso.onmicrosoft.com"
                            };
                            var existingUserList = await context.Web.ValidateAndEnsureUsersAsync(userList);
                        }`
Any idea?

**Comment by jansenbe** [comment, marked as answer]
@AbuAmmar072001 : that method only exists in our nightly builds as it was just recently added. Please install a nightly build.

**Comment by AbuAmmar072001** [comment]
@jansenbe Thank you, it is working fine now.
